rs/execution_environment/src/execution_environment.rs

```diff
@@ -253,6 +253,10 @@ pub struct RoundLimits {
     /// - Wasm execution pushes a new request to the output queue.
     pub subnet_available_memory: SubnetAvailableMemory,
 
+    /// The number of outgoing calls that can still be made across the subnet before
+    /// canisters are limited to their own callback quota.
+    pub subnet_available_callbacks: i64,
```

Comment:
Do we expect this to be <0 ?

Reply:
Yeah, this is a soft cap, so it can be exceeded. We want to accurately track by how much we exceeded it, in case we later start increasing the supply when executing callbacks.

This is something that this PR does not do. Which is OK for the subnet-wide soft cap (it's a porous boundary anyway), but may not be fine for the rather limited canister quota: we probably want canisters to be able to make more downstream calls as soon as they handled a few responses, not only in the next DSM round.

I'll be right back.

Reply:
Never mind, I'm stupid. The "available canister quota" is implicitly updated after every message execution, as we compute it before we execute the next message (as `quota - callbacks.len()`).

Not updating the subnet-wide usage accurately is OK, since there is no guarantee above the 50 callback quota, particularly when the subnet is close to the limit. So we could do it if we wanted to be exceedingly accurate, but it would mean extra complexity. And we're already quite generous fully "replicating" the callback pool across every execution thread to begin with, so it balances out.